roles/satellite-clone/tasks/main.yml

```diff
@@ -279,10 +285,15 @@
       create: yes
       line: "---\n:foreman:\n  :username: admin\n  :password: changeme"
 
-  - name: Run installer upgrade to match latest z-version
+  - name: Upgrade to latest z-version with satellite-installer
     command: 'satellite-installer --upgrade {{ satellite_upgrade_options | default("") }}'
     environment:
       HOSTNAME: "{{ hostname }}"
+    when: satellite_version is version("6.8", "<=")
+
+  - name: Upgrade to latest z-version with satellite-maintain
+    command: 'satellite-maintain upgrade run --target-version={{ satellite_version }}.z --assumeyes'
```

Comment:
Will this work as far back as 6.5? If so I can remove the above equivalent which used satellite-installer

Reply:
Hi @jturel , great question!

As you saw, `--upgrade` was removed (with https://github.com/theforeman/foreman-installer/pull/541/commits/3c60c745fca4a98cf6e04ebbb3d7b8462373e75a ). In fact, at that point it did nothing except print a notice that `--upgrade` is deprecated since we automatically run any pending db migrations (since https://github.com/theforeman/foreman-installer/pull/470/commits/1cc2378a77c73a544de074d420140f274965ce09 ) was included with Foreman 2.1 (corresponding to Satellite 6.8)

IMO, rather than making the change to use satellite-maintain here with 6.9 and later versions, because satellite-maintain introduces a lot of additional overhead (running various pre-upgrade checks, unlocking packages and updating them, etc) which is intended to simplify the user experience of upgrading for customers, but doesn't have much benefit in this specific context, I would simply make the change to remove `--upgrade` from the task here, but include it in `satellite_upgrade_options` only `when: satellite_version is version("6.7", "<=")`

In other words, logically just running the installer still makes sense at this point, we just need to no longer pass `--upgrade` to it as that was deprecated in Satellite 6.8 and has been removed in Satellite 6.9.

Reply:
Slightly different solution: why don't we add `--upgrade` to `satellite_upgrade_options` for 6.5,6.6,6.7 and leave it out in 6.8+? Removes the need for doing wild `satellite_version` checks here in the playbook as we already `satellite_upgrade_options`?

Reply:
Thanks @wbclark & @evgeni - gonna update the vars file to prevent adding back a satellite_version check

Reply:
Updated!

Reply:
Nice!